src/Entity/Fileset.php

```diff
@@ -255,4 +273,63 @@ public function setZipFilePath(string $zipFilePath): void
     {
         $this->zipFilePath = $zipFilePath;
     }
+
+    /**
+     * Getter for zip file size.
+     *
+     * @return string|null
+     */
+    public function getZipFileSize(): ? string
+    {
+        return $this->zipFileSize;
+    }
+
+    /**
+     * Setter for zip file size.
+     *
+     * @param string $zipFileSize Zip file size on disk.
+     *
+     * @return void
+     */
+    public function setZipFileSize(string $zipFileSize): void
+    {
+        $this->zipFileSize = $zipFileSize;
+    }
+
+    /**
+     * Getter for zip file hash value.
+     *
+     * @return string|null
+     */
+    public function getZipFileSha256Hash(): ? string
+    {
+        return $this->zipFileSha256Hash;
+    }
+
+    /**
+     * Setter for zip file hash.
+     *
+     * @param string $zipFileSha256Hash Hash value for the zip file.
+     *
+     * @return void
+     */
+    public function setZipFileSha256Hash(string $zipFileSha256Hash): void
+    {
+        $this->zipFileSha256Hash = $zipFileSha256Hash;
+    }
+
+    /**
+     * Check if zip file exists for the fileset.
+     *
+     * @return bool
+     */
```

Comment:
Would this be more correct with an OR across all 3 attribute instead of the AND? I know they're all rather atomic, so maybe this is more of an academic argument, but I think presence of any of these fields means it exists.

Reply:
True, but we want all these fields to be populated, and it gets populated by the same message handler class ZipDatasetFilesHandler class.